package.json

```diff
@@ -24,25 +34,33 @@
   "bugs": {
     "url": "https://github.com/pengx17/monaco-yaml/issues"
   },
+  "dependencies": {
+    "js-yaml": "^3.12.0",
+    "prettier": "^1.19.1",
```

Comment:
I am a little worried if it is an overkill to include prettier to a web-oriented lib. Can you estimate the impact to the dist bundles with or without it?

Reply:
We only import [prettier/standalone](https://prettier.io/docs/en/browser.html) which is meant to be used in the browser but I meant to make this an optional dependency like it is in [yaml-language-server](https://github.com/redhat-developer/yaml-language-server/blob/bb194cc75a141898a272a287ae884e43b54c5c10/package.json#L40-L42). So i'll just fix that.

Reply:
Not sure require('prettier/standalone') will work properly in a esm bundle since it is a commonjs syntax? Do you mean the user should config their requirejs context to make it work properly? It it is the case, I think we should have a fallback to make it work for most users.

Reply:
TSC will handle that?

Reply:
TSC will not handle it. It will just emit `require('prettier/standalone')` as it is in the lib for amd/esm. At runtime in user's browser it will use require.js to load the "external" resource.
Here is a pr to fix the demo formatter in this pr.
https://github.com/kristenmills/monaco-yaml/pull/2

Reply:
Cool i merged in your PR to mine. I'm not nearly as familiar with the require.js stuff but it makes sense to me.

Reply:
IMO Prettier could be a universal formatter solution for monaco-editor, not for monaco-yaml plugin. We should still use the old way to format YAML documents, until monaco-editor has buitin solution for prettier.

Reply:
I only brought it in because it's included in yaml-language-server.

Reply:
They have different use case. yaml-language-server was implemented for nodejs environment, but not for the browser. If we want to bring it to monaco-editor, we may think another solution instead.

Reply:
why do we need a different solution? Prettier standalone is meant for browser environments

Reply:
I think we could ship it for now. Will open a new issue to fix the formatter in another pr.

Reply:
> why do we need a different solution? Prettier standalone is meant for browser environments

1. the way to load prettier/parser-yaml (it is different for esm/umd for the moment)
2. the size impact
3. prettier could be loaded in another plugin but the way of loading it may be different. We could not make a solid assumption for now

I think an acceptable solution for prettier is to wrap it into a standalone worker.